cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricValue.java

```diff
@@ -24,12 +24,64 @@
   private MetricType type;
   private long value;
 
+  // following fields to support distribution aka histogram/event metrics
+  /*
+  Based on https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#Exponential
+   */
+  public static final int GROWTH_FACTOR = 2;
+
+  // Total buckets = NUM_FINITE_BUCKETS +2 which will be 64. A long number
+  // maybe used as a mask to indicate which buckets have non zero bucket counts
+  public static final int NUM_FINITE_BUCKETS = 62;
+
+  /**
+   * Based on https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#Distribution
+   *
+   * bucketCounts stores counts of buckets with non zero values.
+   */
+  private long[] bucketCounts;
+
+  // Most of the buckets have zero counts. For efficiency, bucketCounts only
+  // stores non zero bucket count values. The bucketMask stores which
+  // buckets have non-zero bucketCounts.
+  private long bucketMask = 0;
+
+  /**
+   * The number of values in the population. Must be non-negative.
+   * This value must equal the sum of the values in bucketCounts
+  */
+  private long count;
+
+  /**
+   * The arithmetic mean of the values in the population. If count is zero then this field must be zero.
+   */
+  private double mean;
```

Comment:
Please use sum. It's generally much easier to do aggregation with sum and count and calculate mean as needed

Reply:
I think calculated values shouldn't be stored in this class. This class is for carrying the raw numbers. The calculated stats can be done at the metrics writer end.

Reply:
Well, after bucketing we don't have exact sum. May be it makes sense to have it still to get more precise mean

Reply:
Maintaining sum. Yes, reason to store the mean (or sum now) is since the actual values are not present.

Reply:
Changing back to mean. Sum has the risk of overflowing.

Reply:
Mean accumulates error because adding another value is some additional multiplication / division. If there is a risk of overflowing, use proper type (e.g. double) for sum

Reply:
With aggregation in mind, I agree sum which is double is better. Changed it.